src/OrleansAWSUtils/Providers/Storage/DynamoDBStorage.cs

```diff
@@ -0,0 +1,415 @@
+            serviceId = providerRuntime.ServiceId.ToString();
+
+            if (config.Properties.ContainsKey(ServicePropertyName))
+                service = config.Properties[ServicePropertyName];
+
+            if (config.Properties.ContainsKey(AccessKeyPropertyName))
+                accessKey = config.Properties[AccessKeyPropertyName];
+
+            if (config.Properties.ContainsKey(SecretKeyPropertyName))
+                secretKey = config.Properties[SecretKeyPropertyName];
+
+            if (config.Properties.ContainsKey(ReadCapacityUnitsPropertyName))
+                readCapacityUnits = int.Parse(config.Properties[ReadCapacityUnitsPropertyName]);
+
+            if (config.Properties.ContainsKey(WriteCapacityUnitsPropertyName))
+                writeCapacityUnits = int.Parse(config.Properties[WriteCapacityUnitsPropertyName]);
+
+            if (config.Properties.ContainsKey(TableNamePropertyName))
+                tableName = config.Properties[TableNamePropertyName];
+
+            isDeleteStateOnClear = config.Properties.ContainsKey(DeleteOnClearPropertyName) &&
+                "true".Equals(config.Properties[DeleteOnClearPropertyName], StringComparison.OrdinalIgnoreCase);
+
+            Log = providerRuntime.GetLogger("Storage.AWSDynamoDBStorage." + id);
+
+            var initMsg = string.Format("Init: Name={0} ServiceId={1} Table={2} DeleteStateOnClear={3}",
+                Name, serviceId, tableName, isDeleteStateOnClear);
+
+            if (config.Properties.ContainsKey(UseJsonFormatPropertyName))
+                useJsonFormat = "true".Equals(config.Properties[UseJsonFormatPropertyName], StringComparison.OrdinalIgnoreCase);
+
+            this.jsonSettings = SerializationManager.UpdateSerializerSettings(SerializationManager.GetDefaultJsonSerializerSettings(), config);
+
+            initMsg = String.Format("{0} UseJsonFormat={1}", initMsg, useJsonFormat);
+
+            Log.Info(ErrorCode.StorageProviderBase, "AWS DynamoDB Provider: {0}", initMsg);
+
+            tableDataManager = new GrainStateTableDataManager(tableName, accessKey, secretKey, service, readCapacityUnits, writeCapacityUnits, Log);
+            return tableDataManager.InitTableAsync();
+        }
+
+        // Internal method to initialize for testing
+        internal void InitLogger(Logger logger)
+        {
+            Log = logger;
+        }
+
+        /// <summary> Shutdown this storage provider. </summary>
+        /// <see cref="IProvider.Close"/>
+        public Task Close()
+        {
+            tableDataManager = null;
+            return TaskDone.Done;
+        }
+
+        /// <summary> Read state data function for this storage provider. </summary>
+        /// <see cref="IStorageProvider.ReadStateAsync"/>
+        public async Task ReadStateAsync(string grainType, GrainReference grainReference, IGrainState grainState)
+        {
+            if (tableDataManager == null) throw new ArgumentException("GrainState-Table property not initialized");
```

Comment:
I'm not sure this is needed, here and in other places. even then, it's not an `ArgumentException`